### ISSUE TITLE
What's a Program to Do?

### ISSUE THREAD

**Issue by Daniel061** [Daniel061/Sophie6.1#30]
If you were faced with the following statement:

"It went up the hill and didn't come back down. Where did it go?"

What information would you require to be able to answer this question?

Please be specific and break it down.

**Comment by mackenzieankrom** [comment]
I would need the following information:
- What is the "top" of a hill.
- What is the meaning of "where"
- Does the program know the meaning of the word "go"?

**Comment by Daniel061** [comment]
Let me rephrase the question;

 It got all tangled up.

In this context, what would you need to answer the question?

**Comment by mackenzieankrom** [comment]
I believe that the program would ask "What is it?" or would ask "is it the subject?"

**Comment by Daniel061** [comment]
I'd rather the program not need to ask these questions. 

If I said that to you, what would you use to answer?

**Comment by mackenzieankrom** [comment]
I would assign the adjective "tangled" to the pronoun "it", to begin with.

**Comment by Daniel061** [comment]
Would you look to past context to apply the association?

**Comment by mackenzieankrom** [comment]
yes, for example if you had said "the cat is blue" and I knew we were talking about a cat, then if you said "it got all tangled up" I would assume we were still talking about the cat.

**Comment by Daniel061** [comment]
ok, I see. but What order?
If I said , "The dog was running" and then "The cat followed" and then
"It got all tangled up."
Would you choose cat to associated tangled to instead of dog based on order of occurrence?

**Comment by mackenzieankrom** [comment]
I would choose cat because it was the most recent subject.

**Comment by Daniel061** [comment]
So humans are sequential machines when no further clues are given.

**Comment by mackenzieankrom** [comment]
Yes, I would say so.

**Comment by Daniel061** [comment]
What clues would tell you to choose the latter subject?

**Comment by mackenzieankrom** [comment]
Recency, it was the last subject to be mentioned so it makes logical sense to choose it.

**Comment by Daniel061** [comment]
If I said "The fast dog ran up the hill and the cat followed." Then I said he got tangled.
Which animal would I be talking about?

**Comment by mackenzieankrom** [comment]
I would still think the cat because you mentioned it second.

If you wanted the program to assume the dog, it would have to ask which animal was the subject.

**Comment by Daniel061** [comment]
Perfect, If there is not enough information to formulate an answer, that's when I would want the program to ask a question. Just as you politely would do.

**Comment by Daniel061** [comment]
People can be ambiguous in their speech.
